iocore/net/P_SSLNetVConnection.h

```diff
@@ -302,26 +303,35 @@ class SSLNetVConnection : public UnixNetVConnection, public ALPNSupport
     return ssl ? SSL_get_cipher_name(ssl) : nullptr;
   }
 
+  void
+  setSSLCurveNID(ssl_curve_id curve_nid)
+  {
+    sslCurveNID = curve_nid;
+  }
+
+  ssl_curve_id
+  getSSLCurveNID() const
+  {
+    return sslCurveNID;
+  }
+
   const char *
   getSSLCurve() const
   {
     if (!ssl) {
       return nullptr;
     }
-
+    ssl_curve_id curve = getSSLCurveNID();
 #ifndef OPENSSL_IS_BORINGSSL
-    int curve_nid = SSL_get_shared_curve(ssl, 0);
-
-    if (curve_nid == NID_undef) {
+    if (curve == NID_undef) {
       return nullptr;
     }
-    return OBJ_nid2sn(curve_nid);
+    return OBJ_nid2sn(curve);
 #else
-    if (uint16_t curve_id = SSL_get_curve_id(ssl); curve_id != 0) {
-      return SSL_get_curve_name(curve_id);
-    } else {
+    if (curve_id == 0) {
```

Comment:
```
./P_SSLNetVConnection.h:331:9: error: use of undeclared identifier 'curve_id'
    if (curve_id == 0) {
        ^
./P_SSLNetVConnection.h:334:31: error: use of undeclared identifier 'curve_id'
    return SSL_get_curve_name(curve_id);
                              ^
2 errors generated.
```

Reply:
Fixed! thanks for checking it against boringssl :)